packages/flutter/lib/src/cupertino/segment_control.dart

```diff
@@ -0,0 +1,191 @@
+      for (int i = 0; i < widget.tabs.length; i++) {
+        SegmentControlItem t = widget.tabs[i];
+        if (t.title == title) {
+          _activeTabIndex = i;
+        }
+      }
+    });
+  }
+}
+
+class _SegmentControlItem extends StatefulWidget {
+  final double _defaultBorderRadius = 3.0;
+
+  final SegmentControlItem cupertinoButtonTab;
+  final _SegmentControlState parent;
+  final _ButtonPlace place;
+  final bool isActive;
+  final Color color;
+  final Color inverseColor;
+
+  _SegmentControlItem(
+      this.parent, this.cupertinoButtonTab, this.place, this.isActive,
+      {this.color = CupertinoColors.activeBlue,
+      this.inverseColor = CupertinoColors.white});
+
+  @override
+  State createState() {
+    return new _SegmentControlItemState(color, inverseColor);
+  }
+}
+
+class _SegmentControlItemState
+    extends State<_SegmentControlItem> {
+  Color color;
+  Color inverseColor;
+  bool tapDown = false;
+
+  _SegmentControlItemState(this.color, this.inverseColor);
+
+  BoxDecoration _boxDecoration(_ButtonPlace place) {
+    BorderRadius radius;
+
+    switch(place) {
+      case _ButtonPlace.start:
+        radius = new BorderRadius.only(
+          topLeft: new Radius.circular(widget._defaultBorderRadius),
+          bottomLeft: new Radius.circular(widget._defaultBorderRadius),
+        );
+        break;
+      case _ButtonPlace.end:
+        radius = new BorderRadius.only(
+          topRight: new Radius.circular(widget._defaultBorderRadius),
+          bottomRight: new Radius.circular(widget._defaultBorderRadius),
+        );
+        break;
+      default:
+        break;
+    }
+
+    BoxDecoration dec = new BoxDecoration(
+      color: widget.isActive ? color : inverseColor,
+      border: place == _ButtonPlace.middle
+          ? new Border(
+              top: new BorderSide(color: tapDown ? inverseColor : color),
+              bottom: new BorderSide(color: tapDown ? inverseColor : color),
+            )
+          : new Border.all(color: tapDown ? inverseColor : color),
+      borderRadius: radius,
+    );
+
+    return dec;
+  }
+
+  void _tabDown() {
+    if (!widget.isActive) {
+      setState(() {
+        tapDown = true;
+        final Color _backupColor = color;
+        color = inverseColor;
+        inverseColor = _backupColor;
+      });
+    }
+  }
+
+  void _tabUp() {
+    if (!widget.isActive) {
+      tapDown = false;
+      final Color _backupColor = color;
+      color = inverseColor;
+      inverseColor = _backupColor;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new GestureDetector(
+      onTapDown: (_) {
+        _tabDown();
+      },
+      onTapUp: (_) {
+       _tabUp();
+      },
+      onTap: () {
+        widget.parent.changeTab(widget.cupertinoButtonTab.title);
+      },
+      child: new Container(
+        decoration: _boxDecoration(widget.place),
+        padding: new EdgeInsets.fromLTRB(20.0, 4.0, 20.0, 4.0),
+        child: new Text(
```

Comment:
I would just let the user pass in a Widget in the SegmentControlItem instead of a string. It's not unimaginable that people might want to have icons instead of text. 

Consider https://developer.apple.com/ios/human-interface-guidelines/controls/segmented-controls/:
"Like buttons, segments can contain text or images"